src/wildcard-service/init.lua

```diff
@@ -50,24 +51,34 @@ function _M.new(options)
     system_developer = extract_service('system-developer', 3000),
   }
 
+  local domain_cache = lrucache.new(resty_env.value('DOMAIN_CACHE_SIZE') or 100)
+  local domain_cache_ttl = resty_env.value('DOMAIN_CACHE_TTL') or 60
+
   return setmetatable({
     options = opts,
     http_client = http_client,
     api_host = api_host,
     services = services,
+    domain_cache = domain_cache,
+    domain_cache_ttl = domain_cache_ttl
   }, mt)
 end
 
-function _M:domain_info(domain)
-  local http_client = self.http_client
-  local url = resty_url.join(self.api_host, '/master/api/domain/', domain)
+-- This function is used as a callback in a timer. The first param is
+-- 'premature'. See https://github.com/openresty/lua-nginx-module#ngxtimerat
+local function get_domain_info(_, domain, api_host, http_client, cache)
+  local url = resty_url.join(api_host, '/master/api/domain/', domain)
 
   local res = http_client.get(url)
 
   ngx.log(ngx.DEBUG, 'domain info for:  ', domain, ' from: ', url, ' status: ', res.status, ' body: ', res.body)
 
   if res.status == 200 then
-    return cjson.decode(res.body)
+    local domain_info = cjson.decode(res.body)
+
+    cache:set(domain, domain_info)
```

Comment:
This should use `domain_cache_ttl` right?

Reply:
Yes. There was a bug. I fixed it and added a test for it:
`it('allows to configure the cache TTL', function()`